### ISSUE TITLE
string option comparison doesn't work with SQLite

### ISSUE THREAD

**Issue by giuliohome** [fsprojects/SQLProvider#478]
### Description

When I do

```fsharp
namespace FunctionalSQLite

open System
open FSharp.Data
open FSharp.Data.Sql

type PriceRec = {RefDate:DateTime; Price:double; Provider: string option}

module TestMe =

    [<Literal>]
    let connStringApp ="Data Source=C:\giulio-vs-so\sqlite\TestMePlease.s3db;Version=3;"

    type appSql = SqlDataProvider< 
                      ConnectionString = connStringApp,
                      DatabaseVendor = Common.DatabaseProviderTypes.SQLITE,
                      IndividualsAmount = 1000,
                      UseOptionTypes = true >  

    FSharp.Data.Sql.Common.QueryEvents.SqlQueryEvent |> Event.add (printfn "Executing SQL: %O")

    let CheckError aPrice =
        let appCtx =appSql.GetDataContext()
        let prices = appCtx.Main.VeryUrgent

        let foundPrice = 
             query {
             for p in prices do 
             where (p.Provider = aPrice.Provider  && p.RefDate = Some aPrice.RefDate)
             exactlyOneOrDefault
             } 
         
        if not(isNull foundPrice) then
            foundPrice.MyValue <- Some (aPrice.Price + 1.1)
          else 
            let newRec = prices.Create()  
            newRec.RefDate <- Some aPrice.RefDate
            newRec.Provider <-  aPrice.Provider
            newRec.MyValue <- Some aPrice.Price

        appCtx.SubmitUpdates()

open TestMe
module VeryUrgent =
    [<EntryPoint>]
    let main argv = 
        let feed = {Price=12.3 ; RefDate=DateTime(2017,10,31); Provider=Some "One" }
        CheckError feed
        0 
```

### Repro steps

Steps required to reproduce the problem

1. Step A
prepare the db

```sql
CREATE TABLE [very_urgent] (
[ref_date] DATE  NULL,
[provider] VARCHAR(20)  NULL,
[my_value] FLOAT  NULL,
PRIMARY KEY ([ref_date],[provider])
)
```
and insert 
```sql
insert into very_urgent values ('2017-10-31','One',1.1)
```
2. Step B
execute the above code

### Expected behavior

code should run fine without error: the where condition sould find when the date field is already there and branch to the update part instead of failing with a duplicate creation

### Actual behavior

A blocking error
```fsharp
System.Data.SQLite.SQLiteException was unhandled
Message: An unhandled exception of type 'System.Data.SQLite.SQLiteException' occurred in System.Data.SQLite.dll
Additional information: constraint failed
UNIQUE constraint failed: very_urgent.ref_date, very_urgent.provider
```
### Related information 

* Used database SQLite Version 3
* .NET Runtime 4.6.1

**Comment by giuliohome** [comment]
in the meantime, setting `UseOptionTypes = false` (plus all the needed code changes) seems to be a work around

**Comment by Thorium** [comment]
Basically `query { ... }` is translated to SQL, so the problem is what is `Some "One"` as SQL. So there are two workarounds here:

1. Don't use option type in your records

```fsharp
type PriceRec = {RefDate:DateTime; Price:double; Provider: string option}
//vs
type PriceRec = {RefDate:DateTime; Price:double; Provider: string}
```

2. Make a temp variable to hold option.Value and use temp variable inside your query-expression.

**Comment by Thorium** [comment]
I guess this could be easy to fix just using option value, but I don't know if that would be clear to the developer using the system:
```
where (p.Provider = aPrice.Provider)
```
What if non-sql parameter is null/None? Should we select * or only null values?

**Comment by giuliohome** [comment]
Only null values

**Comment by Thorium** [comment]
Yes, but in SQL `where x = null` is invalid comparison resulting to random results (depending on the engine), the valid would be `where x is null`.

Anyways, I think that behaviour would be more logical to the current one, so I'll probably change it.

**Comment by giuliohome** [comment]
the _translation_ to SQL `where x = null` would be incorrect indeed and the correct one must be `where x is null`

**Comment by Thorium** [comment]
Fixed in 1.1.19.
